gpu_script_optimized.sh

```
@@ -0,0 +1,162 @@
+if [ -z "$EBROOTLIBYAML" ]; then
+    echo "Error: EBROOTLIBYAML not set. LibYAML module might not be loaded correctly."
+    exit 1
+fi
+
+echo "GSL Path: $EBROOTGSL"
+echo "LibYAML Path: $EBROOTLIBYAML"
+echo "Host CUDA Path: $CUDA_HOST"
+
+# Setup Container Bindings
+# We bind the project directory (pwd), GSL, LibYAML, and Host CUDA
+export APPTAINER_BIND="$(pwd),${EBROOTGSL},${EBROOTLIBYAML},${CUDA_HOST}:/host-cuda"
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)